### ISSUE TITLE
[Issue]: searchHints API Endpoint sometimes return bad request when some characters is in the URL

### ISSUE THREAD

**Issue by tobycm** [jellyfin/jellyfin#7883]
### Please describe your bug

I am developing with Jellyfin API
Let's look at this endpoint: [/search/hints](http://swagger.emby.media/?staticview=true#/SearchService/getSearchHints)

I can authenticate with Jellyfin just fine, try searching some media and success doing so.

Demo:

### Jellyfin Version

10.7.7

### Environment

```markdown
- OS: Server: ARM Raspbian Linux, Client: Windows
- Virtualization: None
- Clients: Jellyfin Windows
- Browser: None
- FFmpeg Version: i don't know how to find this
- Playback Method: None
- Hardware Acceleration: None
- Plugins: AudioDB, AutoOrganize, MusicBrainz, OMDb, Playback Reporting, Reports, Webhook
- Reverse Proxy: nginx + Cloudflare, same thing happen on local too
- Base URL: /jellyfin
- Networking: Host
- Storage: local
```

### Jellyfin logs

```shell
[2022-06-11 15:21:50.518 -07:00] [ERR] Error processing request. URL "GET" "/Search/Hints".
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
   at System.SZArrayHelper.get_Item[T](Int32 index)
   at Jellyfin.Api.Controllers.SearchController.GetSearchHintResult(SearchHintInfo hintInfo)
   at System.Linq.Enumerable.SelectListIterator`2.ToArray()
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Jellyfin.Api.Controllers.SearchController.Get(Nullable`1 startIndex, Nullable`1 limit, Nullable`1 userId, String searchTerm, String[] includeItemTypes, String[] excludeItemTypes, String[] mediaTypes, Nullable`1 parentId, Nullable`1 isMovie, Nullable`1 isSeries, Nullable`1 isNews, Nullable`1 isKids, Nullable`1 isSports, Boolean includePeople, Boolean includeMedia, Boolean includeGenres, Boolean includeStudios, Boolean includeArtists)
   at lambda_method1203(Closure , Object , Object[] )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Jellyfin.Server.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Server.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Server.Middleware.IpBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
   at Jellyfin.Server.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Server.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
```

### FFmpeg logs

```shell
None
```

### Please attach any browser or client logs here

None

### Please attach any screenshots here

Demo: https://youtu.be/MazyhB3SHiM

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by nielsvanvelzen** [comment]
FYI our API documentation is available at api.jellyfin.org or directly on your server by going to [serverurl]/api-docs.

**Comment by tobycm** [comment]
Emby and Jellyfin are really similar, I don't see any diffs

Vào 00:23, CN, 12 Th6, 2022 Niels van Velzen ***@***.***> đã
viết:

**Comment by tobycm** [comment]
Is this fixed?

**Comment by crobibero** [comment]
Yes, fixed in #7894 

Don't know why merging a PR sometimes doesn't close the linked issue
